### PR TITLE
add markClaimed action to hasura schema

### PR DESCRIPTION
**Pull request by levity**
## Motivation and Context

Track claims in interaction_events

## Description

interaction_events is only accessible from the backend, so we create a new Hasura action for claims. then we add the same mutation to it as we did for the other vault operations.

### CODE REVIEW

**Comment by levity** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1446** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1446" target="_blank">Graphite</a> 
    * **PR #1444** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1444" target="_blank">Graphite</a> 
      * **PR #1445** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1445" target="_blank">Graphite</a> 
        * **PR #1426** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1426" target="_blank">Graphite</a> 
          * **PR #1455** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1455" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1455?utm_source=stack-comment).

File: api-lib/gql/__generated__/zeus/index.ts [hunk 1]
```diff
@@ -323,7 +323,7 @@ export class GraphQLError extends Error {
   constructor(public response: GraphQLResponse) {
     super('');
     // eslint-disable-next-line no-console
-    console.info(JSON.stringify(response, null, 2));
+    console.info(JSON.stringify(response));
```
Comment (CryptoGraffe) [inline api-lib/gql/__generated__/zeus/index.ts:326]: how do we fix this permanently? I thought our script did it before

File: api-lib/gql/__generated__/zeus/const.ts [hunk 2]
```diff
@@ -5688,6 +5692,9 @@ export const ReturnTypes: Record<string, any> = {
     id: 'Int',
     profile: 'profiles',
   },
+  MarkClaimedOutput: {
+    ids: 'Int',
```
Comment (CryptoGraffe) [inline api-lib/gql/__generated__/zeus/const.ts:5696]: curious about this being plural but being a singular int

**Review by CryptoGraffe — COMMENTED** [review]
This looks ok to me pending my minor comments

File: api-lib/gql/__generated__/zeus/const.ts (hunk 2 above)
Comment (levity) [reply api-lib/gql/__generated__/zeus/const.ts:5696]: oh good catch, this should be an array type

File: api/hasura/actions/_handlers/markClaimed.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      vault: {
+        vault_address,
+        decimals,
+        symbol,
+        simple_token_address,
+        chain_id,
+      },
+      epoch: { circle_id },
+    },
+  } = claim;
```
Comment (topocount) [inline api/hasura/actions/_handlers/markClaimed.ts:83]: this aggressive expansion is nice

**Review by topocount — APPROVED** [review]
I'd appreciate it if you could add stubbed unit tests to assert the correctness of some error states along with some of the more complex branching, but otherwise approve.

File: api-lib/gql/__generated__/zeus/index.ts (hunk 1 above)
Comment (topocount) [reply api-lib/gql/__generated__/zeus/index.ts:326]: @CryptoGraffe and @levity does this line work on your machines? https://github.com/coordinape/coordinape/blob/dadae6d9eaacc42f4e88d2312bbf534dbad69454/scripts/generate_zeus.sh#L43

Comment (topocount) [reply api-lib/gql/__generated__/zeus/index.ts]: @CryptoGraffe fixed here: https://github.com/coordinape/coordinape/pull/1455/commits/64df0c18a57ce2ffc098e796890086a7e5bab0f5

File: api-lib/gql/__generated__/zeus/const.ts (hunk 2 above)
Comment (levity) [reply api-lib/gql/__generated__/zeus/const.ts:5696]: idunno what's going on here, but the zeus index.ts files have the correct type:
```
['MarkClaimedOutput']: {
  ids: Array<number>;
};
```

File: api-lib/gql/__generated__/zeus/const.ts [hunk 4]
```diff
@@ -287,6 +288,7 @@ export const AllTypesProps: Record<string, any> = {
     hash: 'String_comparison_exp',
     name: 'String_comparison_exp',
     read_circle: 'Boolean_comparison_exp',
+    read_discord: 'Boolean_comparison_exp',
```
Comment (crabsinger) [inline api-lib/gql/__generated__/zeus/const.ts:291]: @levity these seem like erroneous additions to the types... ? I don't see any reference to `read_discord` anywhere else in the codebase, and a local `yarn generate` on my machine removes these additions as well. Are these intentional? Do you perhaps have an extra column in your local db that is generating these?

**Comment by levity** [comment]
Huh. I had assumed they were for the API and had been omitted accidentally,
but I guess it’s the other way around; they were removed intentionally but
I’d already added them to my local db? Anyway feel free to remove

**Pull request merged**